bdikit/functional_api.py

```diff
@@ -0,0 +1,138 @@
+    GPT = ("gpt", GPTAlgorithm)
+
+    def __init__(
+        self, method_name: str, method_class: Type[BaseColumnMappingAlgorithm]
+    ):
+        self.method_name = method_name
+        self.method_class = method_class
+
+    @staticmethod
+    def get_instance(method_name: str) -> BaseColumnMappingAlgorithm:
+        methods = {
+            method.method_name: method.method_class for method in ColumnMappingMethod
+        }
+        try:
+            return methods[method_name]()
+        except KeyError:
+            names = ", ".join(list(methods.keys()))
+            raise ValueError(
+                f"The {method_name} algorithm is not supported. "
+                f"Supported algorithms are: {names}"
+            )
+
+
+def match_columns(
+    source: pd.DataFrame,
+    target: Union[str, pd.DataFrame] = "gdc",
+    method: str = ColumnMappingMethod.SIMFLOOD.name,
+) -> pd.DataFrame:
+    """
+    Performs schema mapping between the source table and the given target. The target
+    either is a DataFrame or a string representing a standard data vocabulary.
+    """
+    if isinstance(target, str):
+        target_table = _load_table_for_standard(target)
+    else:
+        target_table = target
+
+    matcher_instance = ColumnMappingMethod.get_instance(method)
+    matches = matcher_instance.map(source, target_table)
+
+    return pd.DataFrame(matches.items(), columns=["source", "target"])
+
+
+def _load_table_for_standard(name: str) -> pd.DataFrame:
+    """
+    Load the table for the given standard data vocabulary. Currently, only the
+    GDC standard is supported.
+    """
+    if name == "gdc":
+        return pd.read_csv(GDC_DATA_PATH)
+    else:
+        raise ValueError(f"The {name} standard is not supported")
+
+
+def top_matches(
+    source: pd.DataFrame,
+    columns: Optional[List[str]] = None,
+    target: Union[str, pd.DataFrame] = "gdc",
+    top_k: int = 10,
+) -> pd.DataFrame:
+    """
+    Returns the top-k matches between the source and target tables.
+    """
+
+    if isinstance(target, str):
+        target_table = _load_table_for_standard(target)
+    else:
+        target_table = target
+
+    if columns is not None and len(columns) > 0:
+        selected_columns = source[columns]
+    else:
+        selected_columns = source
+
+    model_path = get_cached_model_or_download("cl-reducer-v0.1")
+    api = ContrastiveLearningAPI(model_path=model_path, top_k=top_k)
+    _, scopes_json = api.get_recommendations(selected_columns, target=target_table)
+
+    dfs = []
+    for scope in scopes_json:
+        matches = pd.DataFrame(
+            scope["Top k columns"], columns=["matches", "similarity"]
+        )
+        matches["source"] = scope["Candidate column"]
+        matches = matches[["source", "matches", "similarity"]]
+        dfs.append(matches.sort_values(by="similarity", ascending=False))
+
+    return pd.concat(dfs, ignore_index=True)
+
+
+def materialize_mapping(
```

Comment:
I would suggest adding some initial documentation on the params/returns, as this function will become more complex soon.

Reply:
Added documentation in the last PR.